pkg/virtualization/core/virtualsystem/virtualmachine.go

```diff
@@ -883,6 +891,49 @@ func (vm *VirtualMachine) GetResourceAllocationSettingData(rtype v2.ResourcePool
 	return
 }
 
+func (vm *VirtualMachine) GetResourceAllocationSettingDataBySubType(resourceSubType string) (col *v2.CIM_ResourceAllocationSettingData, err error) {
+	settings, err := vm.GetVirtualSystemSettingData()
+	if err != nil {
+		return
+	}
+	defer settings.Close()
+
+	rasdcol, err := settings.GetAllRelated("CIM_ResourceAllocationSettingData")
+	if err != nil {
+		return
+	}
+	defer rasdcol.Close()
+
+	for _, ins := range rasdcol {
+		rasd, err1 := v2.NewCIM_ResourceAllocationSettingDataEx1(ins)
```

Comment:
Better to declare rasd and use err instead of err1.

Reply:
I had to declare more vars than I thought to do this, I think I'll leave it as is to avoid the code being too verbose